components/merkledb/src/backends/rocksdb.rs

```diff
@@ -108,21 +108,38 @@ impl RocksDB {
     ///
     /// [RocksDB docs]: https://github.com/facebook/rocksdb/wiki/Checkpoints
     pub fn create_checkpoint<T: AsRef<Path>>(&self, path: T) -> crate::Result<()> {
-        let checkpoint = Checkpoint::new(&*self.db)?;
+        let checkpoint = Checkpoint::new(&*self.db.read().expect("Couldn't get read lock to DB"))?;
         checkpoint.create_checkpoint(path)?;
         Ok(())
     }
 
+    fn cf_exists(&self, cf_name: &str) -> bool {
+        self.db
+            .read()
+            .expect("Couldn't get read lock to DB")
+            .cf_handle(cf_name)
+            .is_some()
+    }
+
+    fn create_cf(&self, cf_name: &str) -> crate::Result<()> {
+        self.db
+            .write()
+            .expect("Couldn't get write lock to DB")
+            .create_cf(cf_name, &self.options.into())
+            .map_err(Into::into)
+    }
+
     fn do_merge(&self, patch: Patch, w_opts: &RocksDBWriteOptions) -> crate::Result<()> {
         let mut batch = WriteBatch::default();
         for (resolved, changes) in patch.into_changes() {
-            let cf = match self.db.cf_handle(&resolved.name) {
-                Some(cf) => cf,
-                None => self
-                    .db
-                    .create_cf(&resolved.name, &self.options.into())
-                    .unwrap(),
-            };
+            let cf_exists = self.cf_exists(&resolved.name);
+
+            if !cf_exists {
```

Comment:
Maybe just `if !self.cf_exists(&resolved.name)`, without additional variable?

Reply:
Also: if column family doesn't exist, a lock will be obtained twice in a row. Maybe it worth creating a method `create_cf_if_missing`?

Reply:
I thought about such approach. But I've made so because in the `cf_exists` method we acquire a read lock what is much chipper then a write lock, which `create_cf_if_missing` requires. 
About additional variable. It lets us to release read lock which we acquire in `cf_exists` method.

Reply:
About `create_cf_if_missing` -- makes sense, agree.

>It lets us to release read lock 

Hm. Isn't lock released on `drop`, which will be invoked once value is returned from `cf_exists`?

Reply:
Yeah. It should be ok. You're right. 👍🏻